tests/Benchmarks/System.Binary.Base64/Stiching.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using BenchmarkDotNet.Attributes;
+using Base64Decoder = System.Buffers.Text.Base64; // This name problematic, since Base64 is part of namespace.
+
+namespace System.Binary.Base64.Benchmarks
+{
+    public class Stiching
+    {
+        [Params(1_000, 5_000, 10_000, 20_000, 50_000)]
+        public int InputBufferSize;
+
+        private static byte[] _source;
+        private static byte[] _alignedSource1;
+        private static byte[] _alignedSource2;
+        private static byte[] _notAlignedSource1;
+        private static byte[] _notAlignedSource2;
+        private static byte[] _destination;
+        private static byte[] _expected;
+        private static int _bytesConsumed;
```

Comment:
Not sure if we need the bytesConsumed/bytesWritten. If you want to keep the out parameters, they can be locally declared within the benchmark.

In general, variables that are used only during the setup can remain local variables. I think we should only have static variables for things that are used in the actual benchmark.

Reply:
The reason why I make this as a field, is I thought that *potentiall* some code could be optimized away. I'm not sure what is correct way, so I choose to have conservative way. And I don't even sure that it *would* be actually optimized.

Reply:
@ahsonkhan optimization indeed is mindset. I just have to add function with both options and compare instead of speculate of possibilities. I test it and it don't do any difference.

Reply:
```
                             Method | InputBufferSize | stackSize |     Mean |     Error |    StdDev | Scaled | ScaledSD |
----------------------------------- |---------------- |---------- |---------:|----------:|----------:|-------:|---------:|
                   StichingRequired |            1000 |        10 | 819.2 ns |  4.360 ns |  3.641 ns |      ? |        ? |
            StichingRequiredAnonVar |            1000 |        10 | 818.8 ns |  5.102 ns |  4.773 ns |      ? |        ? |
                                    |                 |           |          |           |           |        |          |
                   StichingRequired |            1000 |        32 | 821.7 ns |  4.944 ns |  4.624 ns |      ? |        ? |
            StichingRequiredAnonVar |            1000 |        32 | 818.6 ns |  5.790 ns |  5.416 ns |      ? |        ? |
                                    |                 |           |          |           |           |        |          |
                   StichingRequired |            1000 |        50 | 833.0 ns |  4.612 ns |  4.314 ns |      ? |        ? |
            StichingRequiredAnonVar |            1000 |        50 | 827.3 ns |  4.840 ns |  3.779 ns |      ? |        ? |
                                    |                 |           |          |           |           |        |          |
                   StichingRequired |            1000 |        64 | 817.7 ns |  3.979 ns |  3.722 ns |      ? |        ? |
            StichingRequiredAnonVar |            1000 |        64 | 816.5 ns |  7.606 ns |  7.114 ns |      ? |        ? |
                                    |                 |           |          |           |           |        |          |
                   StichingRequired |            1000 |       100 | 816.5 ns |  2.387 ns |  2.232 ns |      ? |        ? |
            StichingRequiredAnonVar |            1000 |       100 | 858.9 ns | 17.215 ns | 25.233 ns |      ? |        ? |
                                    |                 |           |          |           |           |        |          |
                   StichingRequired |            1000 |       500 | 874.4 ns | 18.074 ns | 16.906 ns |      ? |        ? |
            StichingRequiredAnonVar |            1000 |       500 | 872.5 ns | 11.198 ns | 10.474 ns |      ? |        ? |
                                    |                 |           |          |           |           |        |          |
                         NoStiching |            1000 |         ? | 854.9 ns |  9.571 ns |  8.485 ns |   1.00 |     0.00 |
                  NoStichingAnonVar |            1000 |         ? | 849.5 ns | 13.218 ns | 12.364 ns |   0.99 |     0.02 |
        StichingRequiredNoThirdCall |            1000 |         ? | 882.7 ns | 20.089 ns | 18.792 ns |   1.03 |     0.02 |
 StichingRequiredNoThirdCallAnonVar |            1000 |         ? | 884.8 ns | 17.167 ns | 17.630 ns |   1.04 |     0.02 |
```

Reply:
If benchmark takes 0ns it means that JIT has optimized it and removed the code (it's an empty method).

The most important thing is to return the result from the benchmark (if there is any), BDN will take care of it and make sure the code does not get eliminated by JIT.